plugins/source-verifier/api.ts

```diff
@@ -1,16 +1,34 @@
-export interface Network {
-  name: 'ropsten' | 'mainnet' | 'rinkeby' | 'goerli'
-  id: number  
-}
-
 export interface VerifiedArtifacts {
   metadata: any
   contracts: any
 }
 
+export interface VerifyResponse {
+  match: 'partial' | 'perfect' | 'no match',
+  success: boolean,
+  message: string
+}
+
+export interface VerifyData {
+  address: string,
+  chain: string | number,
+  files: Record<string, any> 
+}
+
 export interface SourceVerifyApi {
   methods: {
-    fetch(address: string, network: Network): VerifiedArtifacts
-    verify(verifyData: any): void
+    /**
+     * Fetch contract metadata and source files.
+     * @param {string} address - contract address.
+     * @param {string | number} network - should be either network name or network id; both number and string types are accepted.
+     * @return {VerifiedArtifacts} - map containing a metadata.json and source files.
+     */
+    fetch(address: string, network: string | number): VerifiedArtifacts
+    /**
+     * verify a contract. the network is retrieved from the application context.
+     * @param {verifyData} VerifyData - map containing a metadata.json and source files
+     * @return {VerifyResponse}
```

Comment:
In tsdoc there is no reason to specify the type of `@return` & `@param` in the doc as it can get out of sync with the typescript definition. 
The best is to remove the `{ type }` statement & **only** display `@return` & `@param` if you want to explain what it is.